llama/llama.go

```diff
@@ -680,3 +683,27 @@ func (s *SamplingContext) Sample(llamaContext *Context, idx int) int {
 func (s *SamplingContext) Accept(id int, applyGrammar bool) {
 	C.gpt_sampler_caccept(s.c, C.llama_token(id), C.bool(applyGrammar))
 }
+
+// KvCacheTypeFromStr converts a string cache type to the corresponding GGML type value
+func KvCacheTypeFromStr(s string) C.enum_ggml_type {
+	switch s {
+	case "f32":
+		return C.GGML_TYPE_F32
+	case "f16":
+		return C.GGML_TYPE_F16
+	case "q8_0":
+		return C.GGML_TYPE_Q8_0
+	case "q4_0":
+		return C.GGML_TYPE_Q4_0
+	case "q4_1":
+		return C.GGML_TYPE_Q4_1
+	case "iq4_nl":
+		return C.GGML_TYPE_IQ4_NL
+	case "q5_0":
+		return C.GGML_TYPE_Q5_0
+	case "q5_1":
+		return C.GGML_TYPE_Q5_1
```

Comment:
Do we want to support these in this PR? It would be best to choose a best 4-bit case that maximizes compatibility and performance

Reply:
I personally think it would be good to have as much as llama.cpp supports to maximize ollamas potential as a full fledged wrapper, we shouldn't cripple it, i have also found great potential in q5_1 and iq4, the i quants really tend to do better than plain q and for users with low vram those two will be important, as q4_0 can have a bit of loss and q8 can still be a little heavy on some models. But that is just my two cents for whatever it is worth.

Reply:
As mentioned a few other times, I would not recommend using q4_0 for the k/v cache, maybe leave it there for advanced users that don't mind the quality loss for their use case - but absolutely would not recommend using it by default.

As for the other quant types I think they're good to have as it aligns with llama.cpp and they're better than q4_0, but happy to remove them and just keep q8_0 and maybe q4_0?

Reply:
Removed support for anything other than q8_0 and q4_0